### PR TITLE
Tweaks to allow passing a string array when p/invoking

### PR DESCRIPTION
**Pull request by BrandonT42**

### CODE REVIEW

**Comment by brandonlehmann** [comment]
You used an out of date copy. Please rebase and PR the development branch.

**Comment by brandonlehmann** [comment]
Didn’t mean to close it. Sorry.

File: include/turtlecoin-crypto.h [hunk 1]
```diff
@@ -64,7 +64,7 @@ namespace Core
 
             static std::string tree_hash(const std::vector<std::string> hashes);
             static std::string tree_branch(const std::vector<std::string> hashes);
-            static std::string tree_hash_from_branch(const std::vector<std::string> branches, const std::string leaf, const std::string path);
+            static std::string tree_hash_from_branch(const std::vector<std::string> branches, const uint64_t depth, const std::string leaf, const std::string path);
```
Comment (brandonlehmann) [inline include/turtlecoin-crypto.h]: No need to pass `depth` as it's actually `size()` in all use cases

File: src/turtlecoin-crypto.cpp [hunk 2]
```diff
@@ -339,7 +339,7 @@ namespace Core
         return Common::podToHex(_branches[0]);
     }
 
-    std::string Cryptography::tree_hash_from_branch(const std::vector<std::string> branches, const std::string leaf, const std::string path)
+    std::string Cryptography::tree_hash_from_branch(const std::vector<std::string> branches, const uint64_t depth, const std::string leaf, const std::string path)
```
Comment (brandonlehmann) [inline src/turtlecoin-crypto.cpp]: Same here and below

File: src/turtlecoin-crypto.cpp [hunk 3]
```diff
@@ -750,13 +748,11 @@ inline void tree_branch(const char* hashes, const uint64_t hashesLength, char* &
     branch = strdup(_branch.c_str());
 }
 
-inline void tree_hash_from_branch(const char* branches, const uint64_t branchesLength, const char* leaf, const char* path, char* &hash)
+inline void tree_hash_from_branch(const char** branches, const uint64_t branchesLength, const uint64_t depth, const char* &leaf, const char* &path, char* &hash)
```
Comment (brandonlehmann) [inline src/turtlecoin-crypto.cpp]: pretty sure that `depth` and `branchesLength` are identical in all use cases and thus the reason there is no reason to pass it.

**Comment by brandonlehmann** [comment]
@BrandonT42 Shall I close this or keep this open?

**Comment by BrandonT42** [comment]
> @BrandonT42 Shall I close this or keep this open?

Kill it. 😈

**Pull request closed without merging**